File: simple.go
package main

import (
	"fmt"
	"time"
)

func longRun(response chan string) {
	time.Sleep(10e9)
	response <- "done"
}

func main() {
	response := make(chan string)
	go longRun(response)
	value := <-response
	fmt.Println(value)
}

File: simple.exs
defmodule Main do

  def longRun(pid) do
    :timer.sleep(10000)
    pid <- {:result, :done}
  end

  def main() do
    spawn(__MODULE__, :longRun, [self])
    receive do
      {:result, value} ->
        IO.puts value
    end
  end

end

Main.main()
